hedera-node/src/main/java/com/hedera/services/ServicesState.java

```diff
@@ -464,8 +475,13 @@ public RecordsRunningHashLeaf runningHashLeaf() {
         return getChild(StateChildIndices.RECORD_STREAM_RUNNING_HASH);
     }
 
-    public MerkleMap<EntityNumPair, MerkleUniqueToken> uniqueTokens() {
-        return getChild(StateChildIndices.UNIQUE_TOKENS);
+    public UniqueTokenMapAdapter uniqueTokens() {
+        final var tokensMap = getChild(StateChildIndices.UNIQUE_TOKENS);
+        return tokensMap.getClass() == MerkleMap.class
```

Comment:
Is there a reason to not just do `tokensMap instanceof MerkleMap`?

Reply:
`instanceof` is a very expensive operation (at least this was true in Java 8) where as the equality check for .getClass() is not as expensive. That was the reason for avoiding instanceof